ecl/hql/hqldesc.cpp

```diff
@@ -113,15 +129,98 @@ static void expandRecordSymbolsMeta(IPropertyTree * metaTree, IHqlExpression * r
     }
 }
 
+void expandScopeMeta(IPropertyTree * meta, IHqlExpression * expr) {
+    if (expr->hasAttribute(virtualAtom))
+        meta->setPropBool("@virtual", true);
+
+    if (expr->hasAttribute(interfaceAtom)){
+        meta->setProp("Type", "interface");
+    }
+    else {
+        meta->setProp("Type", "module");
+    }
+
+    IPropertyTree* scopes = meta->addPropTree("Parents");
+
+    // Walk Attributes to determine inherited scopes
+    ForEachChild(i, expr){
+        IHqlExpression * cur = expr->queryChild(i);
+        IHqlScope * curBase = cur->queryScope();
+        if (curBase) {
+            IPropertyTree* inherited = scopes->addPropTree("Parent");
+            inherited->setProp("@name", str(cur->queryId()));
+            setFullNameProp(inherited, "@ref", cur);
+        }
+    }
+
+    expandScopeSymbolsMeta(meta, expr->queryScope());
+}
 
-void expandSymbolMeta(IPropertyTree * metaTree, IHqlExpression * expr)
+void expandParamMeta(IPropertyTree * meta, IHqlExpression * cur)
+{
+    IPropertyTree * param = NULL;
+    param = meta->addPropTree("Param");
+
+    if (param)
+    {
+        param->setProp("@name", str(cur->queryId()));
```

Comment:
Again, I will add Types in params, fields, etc once we develop the type system representation.